cmd/heznode/main.go

```diff
@@ -317,31 +316,101 @@ func cmdServeAPI(c *cli.Context) error {
 }
 
 func cmdGetAccountDetails(c *cli.Context) error {
-	ethereumNodeURL := c.String(flagEthNodeURL)
-	auctionContractAddressHex := c.String(flagAuctContractAddrHex)
-	accountAddrHex := c.String(flagAccountAddrHex)
-
-	hezClient, err := client.NewHermezClient(ethereumNodeURL, auctionContractAddressHex)
+	accountParam := c.String(flagAccount)
+	const characters = 42
+	var (
+		addr        *ethCommon.Address
+		accountIdxs []common.Idx
+		bjj         *babyjub.PublicKeyComp
+		err         error
+	)
+	matchIdx, _ := regexp.MatchString("^\\d+$", accountParam)
+	if strings.HasPrefix(accountParam, "0x") { //Check ethereum address
+		addr, err = common.HezStringToEthAddr("hez:"+accountParam, "hezEthereumAddress")
+		if err != nil {
+			return err
+		}
+	} else if len(accountParam) > characters { //Check internal hermez account address
+		bjj, err = common.HezStringToBJJ(accountParam, "BJJ")
+		if err != nil {
+			return err
+		}
+	} else if matchIdx { //Check tokenID
+		value, _ := strconv.Atoi(accountParam)
+		accountIdxs = append(accountIdxs, (common.Idx)(value))
+	} else {
+		log.Error("Invalid parameter. Only accepted ethereum address, bjj address or account index.")
+		return nil
+	}
+	_cfg, err := parseCli(c)
 	if err != nil {
-		log.Errorf("Error during Hermez client initialization: %s\n", err.Error())
-		return err
+		return tracerr.Wrap(fmt.Errorf("error parsing flags and config: %w", err))
 	}
-	log.Infof("Connected to Hermez Smart Contracts...")
-	log.Infof("Pulling account info from a coordinator...")
-	accountDetails, err := account.GetAccountInfo(hezClient, accountAddrHex)
+	cfg := _cfg.node
+
+	dbWrite, err := dbUtils.InitSQLDB(
+		cfg.PostgreSQL.PortWrite,
+		cfg.PostgreSQL.HostWrite,
+		cfg.PostgreSQL.UserWrite,
+		cfg.PostgreSQL.PasswordWrite,
+		cfg.PostgreSQL.NameWrite,
+	)
 	if err != nil {
-		log.Errorf("Error obtaining account details. Account: %s - Error: %s\n", accountAddrHex, err.Error())
-		return err
+		return tracerr.Wrap(fmt.Errorf("dbUtils.InitSQLDB: %w", err))
 	}
-	log.Infof("Found %d account(s)", len(accountDetails.Accounts))
-	for index, account := range accountDetails.Accounts {
+	var dbRead *sqlx.DB
```

Comment:
@ARR552 I don't think it's a good idea to open connection to the database in the main file. I would suggest to reuse same method used at API package.

Reply:
Db conexion is created only if the node is runned. In this case the node never starts. Thats why we have to init the db here. @arnaubennassar @mfcastellani and eduS followed the same approche.

Reply:
I see. Thanks for clarifying it.